keras_preprocessing/sequence.py

```diff
@@ -74,7 +77,14 @@ def pad_sequences(sequences, maxlen=None, dtype='int32',
             sample_shape = np.asarray(s).shape[1:]
             break
 
-    x = (np.ones((num_samples, maxlen) + sample_shape) * value).astype(dtype)
+    if isinstance(value, six.string_types) and dtype != object \
+            and not np.issubdtype(dtype, np.str_) \
+            and not np.issubdtype(dtype, np.unicode_):
+        warnings.warn("`dtype` {} is not compatible with `value`'s type: {}\n"
```

Comment:
Shouldn't we throw?
Also, we try to avoid \  when possible. You can create a variable `is_supported = dtype != object` and add more conditions.

Reply:
Good points.
I didn't want to throw and block people that might be using another dtype that is okay even if it fails my check because of something I didn't realize. I'm happy to change it to throw if you'd prefer it.